src/components/Calculator.js

```diff
@@ -1,28 +1,38 @@
-import React from 'react';
+import React, { useState } from 'react';
+import calculate from '../logic/calculate';
 
 function Calculator() {
-  return (
-    <CalculatorUi />
+  const [object, setObject] = useState(
+    {
+      total: 0,
+      next: '',
+      operation: '',
+    },
   );
-}
 
-const CalculatorUi = () => {
+  const clickedBtn = (e) => {
+    setObject(calculate(object, e.target.textContent));
+  };
+
   const digits = ['AC', '+/-', '%', 7, 8, 9, 4, 5, 6, 3, 2, 1, 0, '.'];
 
-  const operators = ['÷', 'X', '-', '+', '='];
+  const operators = ['÷', 'x', '-', '+', '='];
 
   const digitButtons = digits.map((digit) => (
-    <button className="digit-btn" key={digit} type="button">{digit}</button>
+    <button className="digit-btn" key={digit} type="button" onClick={clickedBtn}>{digit}</button>
   ));
 
   const operatorButtons = operators.map((operator) => (
-    <button className="press-btn amber" key={operator} type="button">{operator}</button>
+    <button className="press-btn amber" key={operator} type="button" onClick={clickedBtn}>{operator}</button>
   ));
-
   return (
     <div className="calculator">
-      <div>
-        <input placeholder="0" className="result" />
+      <div className="result">
+        <span>
+          {object.total}
```

Comment:
[OPTIONAL] Since you have already mapped the data from the array into seperate constants as in `digitButtons` and `operatorButtons` it becomes redundant to still use the slice methods to render this data using the slice mehod. Although `map()` will return an array of all the data but react has a careful way of omitting the array literal ([]) when the data are in the dom.

So what i expect you to do was to render the items using the constanst like this.
```
<div>
  {digitButtons}
  {operateButtons}
</div>
```